File: api-server/src/main/java/com/meonjeo/meonjeo/seller/SellerService.java
package com.meonjeo.meonjeo.seller;

import com.meonjeo.meonjeo.auth.CustomUserDetails;
import com.meonjeo.meonjeo.seller.dto.SellerApplyRequest;
import com.meonjeo.meonjeo.seller.dto.SellerProfileResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.*;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service @RequiredArgsConstructor
public class SellerService {
    private final SellerProfileRepository repo;

    // ===== 공용 =====
    public boolean isApprovedSeller(Long userId){
        if (userId == null) return false;
        return repo.existsByUserIdAndStatus(userId, SellerStatus.APPROVED);
    }

    private Long currentUserId(){
        Authentication a = SecurityContextHolder.getContext().getAuthentication();
        if (a == null || a.getPrincipal() == null) return null;
        if (a.getPrincipal() instanceof CustomUserDetails d) return d.getUserId();
        return null;
    }

    // ===== 사용자: 신청/조회 =====
    @Transactional
    public SellerProfileResponse apply(SellerApplyRequest r){
        Long uid = currentUserId();
        if (uid == null) throw new IllegalStateException("UNAUTHORIZED");

        SellerProfile sp = repo.findByUserId(uid).orElseGet(() ->
                SellerProfile.builder().userId(uid).status(SellerStatus.PENDING).build());

        if (sp.getStatus() == SellerStatus.APPROVED) {
            throw new IllegalStateException("ALREADY_APPROVED");
        }
        // REJECTED → 재신청 허용: 상태 PENDING 으로 전환
        sp.setStatus(SellerStatus.PENDING);
        sp.setRejectReason(null);

        sp.setBizNo(nvl(r.bizNo()));
        sp.setShopName(nvl(r.shopName()));
        sp.setOwnerName(nvl(r.ownerName()));
        sp.setAddr(nvl(r.addr()));
        sp.setCategory(nvl(r.category()));
        sp.setPhone(nvl(r.phone()));

        repo.save(sp);
        // fetch join 위해 다시 로드
        return repo.findById(sp.getId()).map(SellerProfileResponse::of).orElseThrow();
    }

    @Transactional(readOnly = true)
    public SellerProfileResponse me(){
        Long uid = currentUserId();
        if (uid == null) throw new IllegalStateException("UNAUTHORIZED");
        return repo.findByUserId(uid).map(SellerProfileResponse::of)
                .orElse(null);
    }

    // ===== 관리자: 목록/상세/승인/거절 =====
    @Transactional(readOnly = true)
    public Page<SellerProfileResponse> adminSearch(SellerStatus status, String q, Pageable pageable){
        return repo.searchForAdmin(status, q, pageable).map(SellerProfileResponse::of);
    }

    @Transactional(readOnly = true)
    public SellerProfileResponse adminGet(Long id){
        return repo.findById(id).map(SellerProfileResponse::of).orElse(null);
    }

    @Transactional
    public SellerProfileResponse approve(Long id, String memo){
        SellerProfile sp = repo.findById(id).orElseThrow();
        sp.setStatus(SellerStatus.APPROVED);
        sp.setRejectReason(null);
        repo.save(sp);
        return SellerProfileResponse.of(sp);
    }

    @Transactional
    public SellerProfileResponse reject(Long id, String reason){
        SellerProfile sp = repo.findById(id).orElseThrow();
        sp.setStatus(SellerStatus.REJECTED);
        sp.setRejectReason(nvl(reason));
        repo.save(sp);
        return SellerProfileResponse.of(sp);
    }

    private static String nvl(String s){ return (s == null || s.isBlank()) ? null : s.trim(); }
}

File: front-server/src/service/sellerService.js
import api from '../config/axiosInstance';

/**
 * 셀러 등록을 신청하거나 재신청합니다.
 * @param {object} payload - 셀러 신청에 필요한 정보 객체
 * @returns {Promise<object>} - 신청 후 반환되는 셀러 프로필 정보
 */
export const applySeller = async (payload) => {
  // 서버 DTO에 없는 값(zipcode 등)은 보내지 않도록 방어
  const body = {
    bizNo: payload.bizNo?.trim(),
    shopName: payload.shopName?.trim(),
    ownerName: payload.ownerName?.trim(),
    addr: payload.addr?.trim(),
    phone: payload.phone?.trim(),
    category: payload.category?.trim(),
  };
  const { data } = await api.post('/api/seller-apply', body);
  return data;
};

/**
 * 현재 로그인한 사용자의 셀러 신청 현황을 조회합니다.
 * @returns {Promise<object>} - 셀러 프로필 정보. 신청 내역이 없으면 null 반환
 */
export const getSellerStatus = async () => {
  try {
    const { data } = await api.get('/api/seller-apply/me');
    return data;
  } catch (error) {
    // 204 No Content 응답을 받았을 경우, 신청 내역이 없는 것으로 처리
    if (error.response && error.response.status === 204) {
      return null;
    }
    // 그 외 에러는 다시 throw
    throw error;
  }
};